packages/core/src/utils/applyBorderRadius.js

```diff
@@ -0,0 +1,49 @@
+function calculateRoundedProps(radius, rounded) {
+  let topRight = 0,
+    bottomRight = 0,
+    bottomLeft = 0,
+    topLeft = 0
+
+  if (!rounded || rounded === 'round') {
+    return radius
+  } else if (rounded === 'top') {
+    topLeft = topRight = radius
+  } else if (rounded === 'right') {
+    topRight = bottomRight = radius
+  } else if (rounded === 'bottom') {
+    bottomRight = bottomLeft = radius
+  } else if (rounded === 'left') {
+    topLeft = bottomLeft = radius
+  } else if (rounded === 'topLeft') {
+    topLeft = radius
+  } else if (rounded === 'topRight') {
+    topRight = radius
+  } else if (rounded === 'bottomRight') {
+    bottomRight = radius
+  } else if (rounded === 'bottomLeft') {
+    bottomLeft = radius
+  }
+
+  return `${topLeft} ${topRight} ${bottomRight} ${bottomLeft}`
+}
+
+export function borderRadiusAttrs({ borderRadius, rounded, theme }) {
```

Comment:
n-b should we add tests for these new functions?

Reply:
yessir, will do in a subsequent PR